packages/ui-react/src/containers/ModalProvider/useModal.ts

```diff
@@ -0,0 +1,45 @@
+import { type RefObject, useContext, useEffect, useRef } from 'react';
+import useEventCallback from '@jwp/ott-hooks-react/src/useEventCallback';
+
+import { ModalContext } from './ModalProvider';
+
+type Params = {
+  open: boolean;
+  onClose?: () => void;
+  modalRef: RefObject<HTMLElement>;
+  containerRef?: RefObject<HTMLElement>;
+};
+
+export const useModal = ({ open, onClose, modalRef, containerRef }: Params) => {
+  const modalId = useRef(String(Math.round(Math.random() * 1_000_000))).current;
```

Comment:
`useOpaqueId()` / `generateId` util?

Reply:
`useOpaqueId` can be used indeed!

Reply:
Nevermind, it fails in unit tests... We should use it only for element IDs

Reply:
(`generateId` could still be used, but it's not important)